got a test failure:

```
=================================== FAILURES ===================================
__________ FacilityAPITestCase.test_device_admin_can_create_facility ___________
self = <django.db.backends.utils.CursorWrapper object at 0x7f8b0f8023c8>
sql = 'SELECT "kolibriauth_collection"."id", "kolibriauth_collection"."name", "kolibriauth_collection"."dataset_id", "kolibr...kolibriauth_facilitydataset"."id") WHERE ("kolibriauth_collection"."kind" = %s AND "kolibriauth_collection"."id" = %s)'
params = ('classroom', 'facility', '43ec85400423b1fbfefc6f75d32a7fc8')
    def execute(self, sql, params=None):
        self.db.validate_no_broken_transaction()
        with self.db.wrap_database_errors:
            if params is None:
                return self.cursor.execute(sql)
            else:
>               return self.cursor.execute(sql, params)
E               psycopg2.ProgrammingError: operator does not exist: text = uuid
E               LINE 1: ...OM "morango_transfersession" U0 WHERE U0."filter" = ("kolibr...
E                                                                            ^
E               HINT:  No operator matches the given name and argument type(s). You might need to add explicit type casts.
.tox/postgres/lib/python3.5/site-packages/django/db/backends/utils.py:64: ProgrammingError
The above exception was the direct cause of the following exception:
self = <kolibri.core.auth.test.test_api.FacilityAPITestCase testMethod=test_device_admin_can_create_facility>
    def test_device_admin_can_create_facility(self):
        new_facility_name = "New Facility"
        self.client.login(
            username=self.superuser.username,
            password=DUMMY_PASSWORD,
            facility=self.facility1,
        )
        self.assertEqual(
            models.Facility.objects.filter(name=new_facility_name).count(), 0
        )
        response = self.client.post(
            reverse("kolibri:core:facility-list"),
            {"name": new_facility_name},
>           format="json",
        )
kolibri/core/auth/test/test_api.py:329: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
.tox/postgres/lib/python3.5/site-packages/rest_framework/test.py:300: in post
    path, data=data, format=format, content_type=content_type, **extra)
.tox/postgres/lib/python3.5/site-packages/rest_framework/test.py:213: in post
    return self.generic('POST', path, data, content_type, **extra)
.tox/postgres/lib/python3.5/site-packages/rest_framework/test.py:238: in generic
    method, path, data, content_type, secure, **extra)
.tox/postgres/lib/python3.5/site-packages/django/test/client.py:416: in generic
    return self.request(**r)
.tox/postgres/lib/python3.5/site-packages/rest_framework/test.py:289: in request
    return super(APIClient, self).request(**kwargs)
.tox/postgres/lib/python3.5/site-packages/rest_framework/test.py:241: in request
    request = super(APIRequestFactory, self).request(**kwargs)
.tox/postgres/lib/python3.5/site-packages/django/test/client.py:501: in request
    six.reraise(*exc_info)
.tox/postgres/lib/python3.5/site-packages/django/utils/six.py:686: in reraise
    raise value
.tox/postgres/lib/python3.5/site-packages/django/core/handlers/exception.py:41: in inner
    response = get_response(request)
.tox/postgres/lib/python3.5/site-packages/django/core/handlers/base.py:187: in _get_response
    response = self.process_exception_by_middleware(e, request)
.tox/postgres/lib/python3.5/site-packages/django/core/handlers/base.py:185: in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
.tox/postgres/lib/python3.5/site-packages/django/views/decorators/csrf.py:58: in wrapped_view
    return view_func(*args, **kwargs)
.tox/postgres/lib/python3.5/site-packages/rest_framework/viewsets.py:116: in view
    return self.dispatch(request, *args, **kwargs)
.tox/postgres/lib/python3.5/site-packages/rest_framework/views.py:495: in dispatch
    response = self.handle_exception(exc)
.tox/postgres/lib/python3.5/site-packages/rest_framework/views.py:455: in handle_exception
    self.raise_uncaught_exception(exc)
.tox/postgres/lib/python3.5/site-packages/rest_framework/views.py:492: in dispatch
    response = handler(request, *args, **kwargs)
kolibri/core/api.py:133: in create
    self.serialize_object(instance.id), status=HTTP_201_CREATED, headers=headers
kolibri/core/api.py:115: in serialize_object
    return self.serialize(queryset.filter(pk=pk))[0]
kolibri/core/api.py:99: in serialize
    list(map(self._map_fields, self._serialize_queryset(queryset) or [])),
.tox/postgres/lib/python3.5/site-packages/django/db/models/query.py:254: in __bool__
    self._fetch_all()
.tox/postgres/lib/python3.5/site-packages/django/db/models/query.py:1121: in _fetch_all
    self._result_cache = list(self._iterable_class(self))
.tox/postgres/lib/python3.5/site-packages/django/db/models/query.py:106: in __iter__
    for row in compiler.results_iter(chunked_fetch=self.chunked_fetch):
.tox/postgres/lib/python3.5/site-packages/django/db/models/sql/compiler.py:841: in results_iter
    results = self.execute_sql(MULTI, chunked_fetch=chunked_fetch)
.tox/postgres/lib/python3.5/site-packages/django/db/models/sql/compiler.py:899: in execute_sql
    raise original_exception
.tox/postgres/lib/python3.5/site-packages/django/db/models/sql/compiler.py:889: in execute_sql
    cursor.execute(sql, params)
.tox/postgres/lib/python3.5/site-packages/django/db/backends/utils.py:64: in execute
    return self.cursor.execute(sql, params)
.tox/postgres/lib/python3.5/site-packages/django/db/utils.py:94: in __exit__
    six.reraise(dj_exc_type, dj_exc_value, traceback)
.tox/postgres/lib/python3.5/site-packages/django/utils/six.py:685: in reraise
    raise value.with_traceback(tb)
```

--- a/kolibri/core/assets/src/state/modules/core/actions.js
+++ b/kolibri/core/assets/src/state/modules/core/actions.js
@@ -372,7 +372,7 @@ export function getFacilityConfig(store, facilityId) {
     return Promise.resolve(redirectBrowser());
   }
   let datasetPromise;
-  if (selectedFacility && typeof selectedFacility.dataset === 'object') {
+  if (selectedFacility && typeof selectedFacility.dataset !== 'object') {
     datasetPromise = Promise.resolve([selectedFacility.dataset]);
   } else {
     datasetPromise = FacilityDatasetResource.fetchCollection({

--- a/kolibri/core/auth/api.py
+++ b/kolibri/core/auth/api.py
@@ -152,7 +152,6 @@ class FacilityDatasetViewSet(ValuesViewset):
         "location",
         "registered",
         "preset",
-        "num_users_in_facility",
     )
 
     field_map = {"allow_guest_access": lambda x: allow_guest_access()}
@@ -334,14 +333,55 @@ class FacilityViewSet(ValuesViewset):
     values = (
         "id",
         "name",
-        "dataset",
         "num_classrooms",
         "num_users",
         "last_synced",
+        # dataset keys
+        "dataset__id",
+        "dataset__learner_can_edit_username",
+        "dataset__learner_can_edit_name",
+        "dataset__learner_can_edit_password",
+        "dataset__learner_can_sign_up",
+        "dataset__learner_can_delete_account",
+        "dataset__learner_can_login_with_no_password",
+        "dataset__show_download_button_in_learn",
+        "dataset__description",
+        "dataset__location",
+        "dataset__registered",
+        "dataset__preset",
     )
+
+    def _map_dataset(facility):
+        dataset = {}
+        dataset["id"] = facility.pop("dataset__id")
+        dataset["learner_can_edit_username"] = facility.pop(
+            "dataset__learner_can_edit_username"
+        )
+        dataset["learner_can_edit_name"] = facility.pop(
+            "dataset__learner_can_edit_name"
+        )
+        dataset["learner_can_edit_password"] = facility.pop(
+            "dataset__learner_can_edit_password"
+        )
+        dataset["learner_can_sign_up"] = facility.pop("dataset__learner_can_sign_up")
+        dataset["learner_can_delete_account"] = facility.pop(
+            "dataset__learner_can_delete_account"
+        )
+        dataset["learner_can_login_with_no_password"] = facility.pop(
+            "dataset__learner_can_login_with_no_password"
+        )
+        dataset["show_download_button_in_learn"] = facility.pop(
+            "dataset__show_download_button_in_learn"
+        )
+        dataset["description"] = facility.pop("dataset__description")
+        dataset["location"] = facility.pop("dataset__location")
+        dataset["registered"] = facility.pop("dataset__registered")
+        dataset["preset"] = facility.pop("dataset__preset")
+        return dataset
 
     field_map = {
         "default": lambda x: Facility.get_default_facility().id == x["id"],
+        "dataset": _map_dataset,
     }
 
     def get_queryset(self, prefetch=True):

--- a/kolibri/core/auth/api.py
+++ b/kolibri/core/auth/api.py
@@ -29,6 +29,8 @@ from rest_framework import filters
 from rest_framework import permissions
 from rest_framework import status
 from rest_framework import viewsets
+from rest_framework.decorators import list_route
+from rest_framework.renderers import JSONRenderer
 from rest_framework.response import Response
 
 from .constants import collection_kinds
@@ -48,6 +50,7 @@ from .serializers import ClassroomSerializer
 from .serializers import FacilityDatasetSerializer
 from .serializers import FacilitySerializer
 from .serializers import FacilityUserSerializer
+from .serializers import FacilityUsersForFacilitiesSerializer
 from .serializers import LearnerGroupSerializer
 from .serializers import MembershipSerializer
 from .serializers import PublicFacilitySerializer
@@ -182,6 +185,20 @@ class FacilityUserViewSet(ValuesViewset):
         "is_superuser": lambda x: bool(x.pop("devicepermissions__is_superuser"))
     }
 
+    @list_route(methods=["get"],)
+    def users_for_facilities(self, request):
+        facility_ids = dict(request.GET)["member_of"]
+
+        # The GET params come as an array with one comma-separated str if
+        # there are more than 1 facility_id being requested
+        if "," in facility_ids[0]:
+            facility_ids = facility_ids[0].split(",")
+
+        users = FacilityUser.objects.filter(facility__in=facility_ids)
+        serialized = FacilityUsersForFacilitiesSerializer(users)
+        return Response(JSONRenderer().render(serialized.data))
+
+    
     def consolidate(self, items, queryset):
         output = []
         items = sorted(items, key=lambda x: x["id"])

--- a/kolibri/core/auth/serializers.py
+++ b/kolibri/core/auth/serializers.py
@@ -75,6 +75,7 @@ class MembershipSerializer(serializers.ModelSerializer):
 
 class FacilityDatasetSerializer(serializers.ModelSerializer):
     allow_guest_access = serializers.SerializerMethodField()
+    num_users_in_facility = serializers.SerializerMethodField()
 
     class Meta:
         model = FacilityDataset
@@ -92,10 +93,36 @@ class FacilityDatasetSerializer(serializers.ModelSerializer):
             "allow_guest_access",
             "registered",
             "preset",
+            "num_users_in_facility",
         )
 
     def get_allow_guest_access(self, instance):
         return allow_guest_access()
+
+    def get_num_users_in_facility(self, instance):
+        facility = Facility.objects.get(dataset_id=instance.id)
+        return FacilityUser.objects.filter(facility=facility).count()
+
+
+class FacilityUsersForFacilitiesSerializer(serializers.Serializer):
+    users = serializers.SerializerMethodField()
+
+    class Meta:
+        fields = "users"
+
+    def get_users(self, users):
+        def sanitize_users(user):
+            return {
+                "id": user.id,
+                "username": user.username,
+                "facility_id": user.facility_id,
+                "is_learner": len(user.get_roles_for_user(user)) == 0,
+                "needs_password": (
+                    user.password == "NOT_SPECIFIED" or not user.password
+                ),
+            }
+
+        return list(map(sanitize_users, users))
 
 
 class FacilitySerializer(serializers.ModelSerializer):

--- a/kolibri/core/auth/serializers.py
+++ b/kolibri/core/auth/serializers.py
@@ -87,16 +87,10 @@ class FacilityDatasetSerializer(serializers.ModelSerializer):
 
 
 class FacilitySerializer(serializers.ModelSerializer):
-    dataset = FacilityDatasetSerializer(read_only=True)
-
     class Meta:
         model = Facility
-        extra_kwargs = {"id": {"read_only": True}, "dataset": {"read_only": True}}
-        fields = (
-            "id",
-            "name",
-            "dataset",
-        )
+        extra_kwargs = {"id": {"read_only": True}}
+        fields = ("id", "name")
 
 
 class PublicFacilitySerializer(serializers.ModelSerializer):

--- a/kolibri/plugins/user/assets/src/modules/signIn/index.js
+++ b/kolibri/plugins/user/assets/src/modules/signIn/index.js
@@ -25,7 +25,7 @@ export default {
 
       FacilityUserResource.getListEndpoint('users_for_facilities', getParams)
         .then(response => {
-          store.commit('SET_SELECTED_FACILITY_USERS', JSON.parse(response.data));
+          store.commit('SET_SELECTED_FACILITY_USERS', response.data);
         })
         .catch(e => {
           console.error(e);

--- a/kolibri/plugins/user/assets/src/modules/signIn/index.js
+++ b/kolibri/plugins/user/assets/src/modules/signIn/index.js
@@ -1,7 +1,11 @@
+import { FacilityUserResource } from 'kolibri.resources';
+
 export default {
   namespaced: true,
   state: {
     hasMultipleFacilities: null,
+    // These users are retrieved for specified facilities
+    usersForSelectedFacilities: [],
   },
   mutations: {
     SET_STATE(state, payload) {
@@ -9,6 +13,23 @@ export default {
     },
     RESET_STATE(state) {
       state.hasMultipleFacilities = null;
+    },
+    SET_SELECTED_FACILITY_USERS(state, payload) {
+      console.log(payload);
+      state.usersForSelectedFacilities = payload;
+    },
+  },
+  actions: {
+    fetchUsersForFacilities(store, payload) {
+      const getParams = { member_of: payload };
+
+      FacilityUserResource.getListEndpoint('users_for_facilities', getParams)
+        .then(response => {
+          store.commit('SET_SELECTED_FACILITY_USERS', JSON.parse(response.data)['users']);
+        })
+        .catch(e => {
+          console.error(e);
+        });
     },
   },
 };

--- a/kolibri/plugins/user/assets/src/modules/signIn/index.js
+++ b/kolibri/plugins/user/assets/src/modules/signIn/index.js
@@ -21,10 +21,11 @@ export default {
   actions: {
     fetchUsersForFacilities(store, payload) {
       const getParams = { member_of: payload };
+      console.log(payload);
 
       FacilityUserResource.getListEndpoint('users_for_facilities', getParams)
         .then(response => {
-          store.commit('SET_SELECTED_FACILITY_USERS', JSON.parse(response.data)['users']);
+          store.commit('SET_SELECTED_FACILITY_USERS', JSON.parse(response.data));
         })
         .catch(e => {
           console.error(e);

--- a/kolibri/plugins/user/assets/src/views/SignInPage.vue
+++ b/kolibri/plugins/user/assets/src/views/SignInPage.vue
@@ -277,9 +277,9 @@
       };
     },
     computed: {
-      ...mapGetters(['facilityConfig', 'selectedFacility', 'isAppContext']),
+      ...mapGetters(['facilityConfig', 'facilities', 'selectedFacility', 'isAppContext']),
       ...mapState(['facilityId']), // backend's default facility on load
-      ...mapState('signIn', ['hasMultipleFacilities']),
+      ...mapState('signIn', ['hasMultipleFacilities', 'usersForSelectedFacilities']),
       ...mapState({
         invalidCredentials: state => state.core.loginError === LoginErrors.INVALID_CREDENTIALS,
         busy: state => state.core.signInBusy,
@@ -289,7 +289,7 @@
       },
       shouldShowUsersList() {
         return (
-          this.usersForCurrentFacility.length <= MAX_USERS_FOR_LISTING_VIEW &&
+          this.facilityConfig.num_users_in_facility <= MAX_USERS_FOR_LISTING_VIEW &&
           this.isAppContext &&
           !this.selectedListUser
         );
@@ -303,9 +303,6 @@
           sug.toLowerCase().startsWith(this.username.toLowerCase())
         );
       },
-      allUsers() {
-        return plugin_data.deviceUsers || [];
-      },
       usernameIsInvalidText() {
         if (this.usernameBlurred || this.formSubmitted) {
           if (this.username === '') {
@@ -317,7 +314,7 @@
         return '';
       },
       usersForCurrentFacility() {
-        return this.allUsers.filter(user => user.facility_id === this.facilityId);
+        return this.usersForSelectedFacilities.filter(user => user.facility_id === this.facilityId);
       },
       usernameIsInvalid() {
         return Boolean(this.usernameIsInvalidText);
@@ -363,6 +360,34 @@
         this.setSuggestionTerm(newVal);
       },
     },
+    created() {
+      // Only get facilities that meet our criteria for listing the users
+      const facilityIdsToFetch = this.facilities
+        .filter(f => f.dataset.num_users_in_facility <= MAX_USERS_FOR_LISTING_VIEW)
+        .map(f => f.id);
+      this.$store.dispatch('signIn/fetchUsersForFacilities', facilityIdsToFetch);
+    },
+    mounted() {
+      /*
+        Chrome has non-standard behavior with auto-filled text fields where
+        the value shows up as an empty string even though there is text in
+        the field:
+          https://bugs.chromium.org/p/chromium/issues/detail?id=669724
+        As super-brittle hack to detect the presence of auto-filled text and
+        work-around it, we look for a change in background color as described
+        here:
+          https://stackoverflow.com/a/35783761
+      */
+      setTimeout(() => {
+        const bgColor = window.getComputedStyle(this.$refs.username.$el.querySelector('input'))
+          .backgroundColor;
+
+        if (bgColor === 'rgb(250, 255, 189)') {
+        }
+      }, 250);
+    },
+          this.autoFilledByChromeAndNotEdited = true;
+        }
     methods: {
       ...mapActions(['kolibriLogin', 'kolibriLoginWithNewPassword', 'clearLoginError']),
       unselectListUser() {

--- a/kolibri/plugins/user/assets/src/views/SignInPage.vue
+++ b/kolibri/plugins/user/assets/src/views/SignInPage.vue
@@ -289,7 +289,7 @@
       },
       shouldShowUsersList() {
         return (
-          this.facilityConfig.num_users_in_facility <= MAX_USERS_FOR_LISTING_VIEW &&
+          this.selectedFacility.num_users <= MAX_USERS_FOR_LISTING_VIEW &&
           this.isAppContext &&
           !this.selectedListUser
         );
@@ -363,7 +363,7 @@
     created() {
       // Only get facilities that meet our criteria for listing the users
       const facilityIdsToFetch = this.facilities
-        .filter(f => f.dataset.num_users_in_facility <= MAX_USERS_FOR_LISTING_VIEW)
+        .filter(f => f.num_users <= MAX_USERS_FOR_LISTING_VIEW)
         .map(f => f.id);
       this.$store.dispatch('signIn/fetchUsersForFacilities', facilityIdsToFetch);
     },

--- a/kolibri/plugins/user/kolibri_plugin.py
+++ b/kolibri/plugins/user/kolibri_plugin.py
@@ -24,23 +24,9 @@ class UserAsset(webpack_hooks.WebpackBundleHook):
 
     @property
     def plugin_data(self):
-        from kolibri.core.auth.models import FacilityUser
-
-        def sanitize_users(user):
-            return {
-                "id": user.id,
-                "username": user.username,
-                "facility_id": user.facility_id,
-                "is_learner": len(user.get_roles_for_user(user)) == 0,
-                "needs_password": (
-                    user.password == "NOT_SPECIFIED" or not user.password
-                ),
-            }
-
         return {
             "oidcProviderEnabled": OIDCProviderHook.is_enabled(),
             "allowGuestAccess": get_device_setting("allow_guest_access"),
-            "deviceUsers": list(map(sanitize_users, FacilityUser.objects.all())),
         }
 
 

